articles/10-tips-for-working-with-node.js/index.md

```diff
@@ -0,0 +1,295 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /engineering-education/10-tips-working-with-nodejs/
+title: 10 Tips for Working with Node.js	
+description: Discover ten of the common tips you need to know from	
+getting started to deploying your first Node.js application.	
+author: joseph-chege
+date: 2020-08-22T00:00:00-10:00	
+topics: [Node.js]	
+excerpt_separator: <!--more-->	
+images:		
+  - url: /engineering-education/10-tips-for-working-with-nodejs/hero.jpg	
+    alt: laptop node.js website
+---
+As a developer getting started with Node.js application development, there a few basic skills that you need to be familiar with as you start to build your first Node.js application.
+
+This article will cover some of the common tips you need to know in order to have a strong background to get started with Node.js application development.
+
+## 1. Understand the Node.js Package Manager (NPM)
+
+NPM is an easy to use package manager for the Node.js platform. It helps to install packages and manage versions and dependencies. NPM helps to install the package you want with just a single command. 
+
+To use NPM, you need Node.js to be installed on your computer. For OS X and Windows, the Node.js installer has an easy to use interface and NPM is included. Developers using Linux need to run the following command in their package manager to set up Node.js and NPM.
+
+```bash
+sudo apt-get update
+
+sudo apt-get install nodejs
+
+sudo apt-get install npm
+```
+
+You can then verify that NPM has been installed by writing the following in your
+terminal `npm -v` or if you realize you have an older version run `npm install npm -g` (for Windows) or `npm install -g npm\@latest` (for Linux and Mac) to update to the latest version of NPM. 
+
+NPM helps to install Node.js modules. For example, you just need to write `npm install` followed by your specified module i.e `npm install express`. 
+
+Once the package is installed you will be able to use `require ()` on your Node.js project and they will run just as if they were built-ins. 
+
+Another important aspect of Npm is dependency management. You can run `npm install` in your root folder and any listed dependency in your package.json (if you have one) will be installed. 
+
+This will help mostly when installing Node.js projects from git repositories. For example, `vow` (a Node.js testing framework) can be installed from git and its single dependency `eye` can automatically be installed i.e.
+
+```bash
+git clone https://github.com/clonehad/vows.git
```

Comment:
The link to the Github repository doesn't work. There's a 404 page not found error.